contracts/BananaAllocator.sol

```diff
@@ -0,0 +1,235 @@
+    function setSyncDelay(uint256 _syncDelay) external onlyOwner {
+        syncDelay = _syncDelay;
+
+        emit SyncDelaySet(_syncDelay);
+    }
+
+    /// @notice A function to add a new allocation
+    /// @param allocationName string referencing the name of an allocation
+    /// @param allocationAmount amount for allocation to recieve
+    /// @param newAllocationAdmins array of addresses to be admins in this specific allocation
+    function addAllocation(
+        string memory allocationName, 
+        uint256 allocationAmount, 
+        address[] memory newAllocationAdmins
+    ) external onlyOwner {
+        totalAllocations += allocationAmount;
+
+        allocationInfo.push(AllocationInfo({
+            allocationName: allocationName,
+            allocationAmount: allocationAmount, 
+            tokensAvailable: 0
+        }));
+
+        uint256 _aid = allocationInfo.length - 1;
+
+        for(uint256 i = 0; i < newAllocationAdmins.length; i++) {
+            allocationInfo[_aid].allocationAdmins[newAllocationAdmins[i]] = true;
+        }
+    
+        emit AllocationAdded(allocationName, allocationAmount, newAllocationAdmins);
+    }
+
+    /// @notice An external function to adjust the amount an allocation has
+    /// @param aid number referencing the applicable allocation
+    /// @param allocationAmount the updated amount to set an allocation to
+    function setAllocation(
+        uint256 aid, 
+        uint256 allocationAmount
+    ) external onlyOwner {
+        uint256 prevAllocationAmount = allocationInfo[aid].allocationAmount;
+
+        allocationInfo[aid].allocationAmount = allocationAmount;
+        if (prevAllocationAmount != allocationAmount) {
+            totalAllocations = totalAllocations.sub(prevAllocationAmount).add(allocationAmount);
+        }
+
+        emit AllocationSet(aid, allocationAmount);
+    }
+
+    /// @notice An external function to add admins to an allocation
+    /// @param aid number referencing the applicable allocation
+    /// @param newAllocationAdmins array of addresses to add to a specific allocation
+    function addAdminsToAllocation(
+        uint256 aid, 
+        address[] memory newAllocationAdmins
+    ) external onlyOwner {
+        for(uint256 i = 0; i < newAllocationAdmins.length; i++) {
+            allocationInfo[aid].allocationAdmins[newAllocationAdmins[i]] = true;
+        }
+
+        emit AdminsAdded(aid, newAllocationAdmins);
+    }
+
+    /// @notice An external function to remove admins from an allocation
+    /// @param aid number referencing the applicable allocation
+    /// @param allocationAdminsToRemove array of addresses to remove from a specific allocation
+    function removeAdminsFromAllocation(
+        uint256 aid, 
+        address[] memory allocationAdminsToRemove
+    ) external onlyOwner {
+        for(uint256 i = 0; i < allocationAdminsToRemove.length; i++) {
+            allocationInfo[aid].allocationAdmins[allocationAdminsToRemove[i]] = false;
```

Comment:
I think it's a good idea to add a require here incase the owner tries to remove admins for the wrong aid or tries to remove an admin they thought was one and wasn't, yet the tx would still succeed. 
`require(allocationInfo[aid].allocationAdmins[allocationAdminsToRemove[i]], "address is not an admin of aid");`

Reply:
Good call, added.